### PR TITLE
Starting CI

### PR DESCRIPTION
**Pull request by rcaneill**
I don't know how to test the CI before merging and doing a fake PR to test it.
If you know how to do (or know a somewhere where they describe it), feel free to do it / tell me (I'll also try to search the answer online whenever I have time).

### CODE REVIEW

**Comment by rcaneill** [comment]
Oh that's magic, github is runing the CI :D, sweet

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python # install poetry
+        source $HOME/.poetry/env
+        poetry install
+    - name: Test with pytest
+      run: |
+        /home/$HOME/.poetry/bin/poetry run pytest gsw_xarray/tests
```
Comment (rcaneill) [inline .github/workflows/ci.yml:36]: Hum, this does not work... It seems that poetry is not found...

Comment (DocOtak) [reply .github/workflows/ci.yml:36]: Poetry seems popular enough that there is probably some gh action we can use to deal with most of it.. I'm getting 500 errors on the gh marketplace so cannot really look

**Comment by rcaneill** [comment]
I am not sure that we need to test buikd / test for both macOS and linux, for 3 python versions.
Github provides 2000 free minutes per month (1 windows minute counts for 2 minutes, 1 macOS minute counts for 10), so I guess that if we reach this limit we can think about reducing the number of tested versions.

**Comment by rcaneill** [comment]
python 3.10 does not pass the tests, it seems to be a pytest error [(see error)](https://github.com/DocOtak/gsw-xarray/runs/5048590445?check_suite_focus=true) but I don't really understand what it is now. I'll search it if I have time today.

File: pyproject.toml [hunk 2]
```diff
@@ -10,7 +10,7 @@ xarray = "^0.20.2"
 gsw = "^3.4.0"
 
 [tool.poetry.dev-dependencies]
-pytest = "^5.2"
+pytest = ">=6.2.5"
```
Comment (rcaneill) [inline pyproject.toml:13]: This is needed for python3.10

**Pull request merged**